public_data/management/commands/build_commune_data.py

```diff
@@ -139,18 +133,20 @@ def __calculate_ocsge_availability(self, city: Commune) -> None:
             ocsge_count_in_city_center_point == available_millesime_in_departement_count
         )
 
-        if has_ocge_coverage:
-            city.ocsge_available = True
+        city.ocsge_available = has_ocge_coverage
 
-    def __calculate_ocsge_first_and_last_millesime(self, city: Commune):
-        queryset = Ocsge.objects.intersect(city.mpoly).distinct("year")
+    def __calculate_ocsge_first_and_last_millesime(self, city: Commune) -> None:
+        queryset = (
+            Ocsge.objects.intersect(city.mpoly).filter(mpoly__contains=city.mpoly.point_on_surface).distinct("year")
```

Comment:
On taperait pas plutôt directement dans le département ?

Reply:
dans public_data.administration class Commune, il y a la méthode : get_ocsge_millesimes()

```suggestion
millesimes = city.get_ocsge_millesimes()
city.last_millesime = max(millesimes)
city.first_millesime = min(millesimes)
```
Je me demande même si tu peux pas sortir ça pour faire un update...

Reply:
Commune.objects.all().update(first_millesime=Min(F("departement__millesimes")), last_millesime=Max(F("departement__millesimes")))

ça marcherait pas ?

Reply:
Et on peut le mettre dans la fonction de setup des départements

Reply:
Ouep bonne idée ! J'ai appliqué ta suggestion à ocsge_availability aussi du coup:
```python
 city.ocsge_available = bool(city.departement.ocsge_millesimes)
```

Reply:
Je viens de voir ta deuxième suggestion. Tu veux faire quoi ?